### DISCUSSION TITLE
sitemap and i18n with app router

### DISCUSSION THREAD

**Discussion by PxlSyl** [vercel/next.js#58490] [Help]
### Summary

How to generate all url in robots.txt for a i18n website?

I made an open source fork of https://github.com/timlrx/tailwind-nextjs-starter-blog and implemented i18next, with all needed features for a translated blog template.

Here's my repo : https://github.com/PxlSyl/tailwind-nextjs-starter-blog-i18n/tree/main

There's a sitemap.ts file in the app folder, I try to modify it in order to get all routes for all locales (SEO improvment) but I don't clearly understand how to do this. 

### Additional information

```js
import { MetadataRoute } from 'next'
import { allBlogs } from 'contentlayer/generated'
import siteMetadata from '@/data/siteMetadata'

export default function sitemap(): MetadataRoute.Sitemap {
  const siteUrl = siteMetadata.siteUrl
  const blogRoutes = allBlogs.map((post) => ({
    url: `${siteUrl}/${post.path}`,
    lastModified: post.lastmod || post.date,
  }))

  const routes = ['', 'blog', 'projects', 'tags'].map((route) => ({
    url: `${siteUrl}/${route}`,
    lastModified: new Date().toISOString().split('T')[0],
  }))

  return [...routes, ...blogRoutes]
}
```

**Comment by kamesdev** [comment]
Did you solve it?
